api/openapi.yaml

```diff
@@ -2342,6 +2692,427 @@ components:
+        - type: object
+          description: "Request for updating existing webhook notification channel."
+          required:
+            - url
+          properties:
+            url:
+              description: Webhook URL where the notification is sent.
+              type: string
+              example: "https://example.com/webhook"
+            customHeaders:
+              description: Custom HTTP headers sent as part of the webhook request.
+              type: object
+              additionalProperties: true
+              nullable: true
+              example:
+                X-CUSTOM-HEADER: "value"
+            signingSecret:
+              description: |
+                Signing secret used for webhook request validation on the receiving end. Automatically generated if not provided.
+
+                Format: `base64` encoded random bytes optionally prefixed with `whsec_`. Recommended size: 24
+              type: string
+              pattern: "^(whsec_)?[a-zA-Z0-9+/=]{32,100}$"
+              nullable: true
+              example: "whsec_S6g2HLnTwd9AhHwUIMFggVS9OfoPafN8"
+
+    NotificationChannelWebhook:
+      allOf:
+        - $ref: "#/components/schemas/GenericNotificationChannel"
+        - type: object
+          description: "Webhook type of notification channel"
+          required:
+            - url
+            - signingSecret
+          properties:
+            url:
+              description: Webhook URL where the notification is sent.
+              type: string
+              example: "https://example.com/webhook"
+            customHeaders:
+              description: Custom HTTP headers sent as part of the webhook request.
+              type: object
+              additionalProperties: true
+              nullable: true
+              example:
+                X-CUSTOM-HEADER: "value"
+            signingSecret:
+              description: |
+                Signing secret used for webhook request validation on the receiving end.
+
+                Format: `base64` encoded random bytes optionally prefixed with `whsec_`. Recommended size: 24
+              type: string
+              pattern: "^(whsec_)?[a-zA-Z0-9+/=]{32,100}$"
+              example: "whsec_S6g2HLnTwd9AhHwUIMFggVS9OfoPafN8"
+
+    NotificationChannelType:
+      description: The type of the notification channel.
+      type: string
+      enum:
+        - WEBHOOK
+      example: WEBHOOK
+
+    GenericNotificationChannelInputs:
+      type: object
+      required:
+        - name
+        - type
+      properties:
+        name:
+          description: User friendly name of the channel.
+          type: string
+          example: "customer-webhook"
+        disabled:
+          description: Whether the channel is disabled or not.
+          type: boolean
+          default: false
+          example: false
+        type:
+          $ref: "#/components/schemas/NotificationChannelType"
+
+    GenericNotificationChannel:
```

Comment:
Adding a `description` with use-case and an `example` would really help first time readers to get the concept quickly.

Reply:
The `GenericNotificationChannel` is a actually a meta object for covering all the common fields across different type of Notification Channels. However I'll update the descriptions and examples for objects that the user directly interacting with.